compiler/src/main/java/org/qbicc/type/definition/classfile/MethodParser.java

```diff
@@ -1542,7 +1544,24 @@ void processNewBlock() {
                         TypeDescriptor desc = getDescriptorOfFieldRef(fieldRef);
                         String name = getNameOfFieldRef(fieldRef);
                         v1 = pop1();
-                        if (v1.getType() instanceof CompoundType ct) {
+                        if (v1 instanceof MemberSelector ms) {
+                            v1 = ms.getInput();
+                            if (v1.getType() instanceof PointerType pt) {
+                                ValueHandle member;
+                                if (pt.getPointeeType() instanceof CompoundType ct) {
+                                    member = gf.memberOf(gf.pointerHandle(v1), ct.getMember(name));
+                                } else if (pt.getPointeeType() instanceof PhysicalObjectType) {
+                                    member = gf.instanceFieldOf(gf.pointerHandle(v1), owner, name, desc);
+                                } else {
+                                    ctxt.getCompilationContext().error(gf.getLocation(), "Invalid field dereference of '%s'", name);
+                                    throw new BlockEarlyTermination(gf.unreachable());
+                                }
+                                push(gf.selectMember(gf.addressOf(member)), false);
```

Comment:
Why rewrap the Value as a SelectMember instead of pushing the address value directly?

Reply:
Because the result of the field selection might in turn be used for another field selection, e.g.

```java
addr_of(fooPtr.sel().bar.baz.zap)
```

and we need it to be impossible for any of the intermediate accesses to be translated into a load.